Ticket1125 device screens dialog

mattclarke opened the pull request:
### Description of work
https://github.com/ISISComputingGroup/IBEX/issues/1125

This adds a dialog for configuring device screens and saving them.

### To test

Open the editor and create/modify some screens and hit "OK" to save.
Confirm that the changes are saved to the screens.xml file in the configuration directory.

**The set default button does nothing at the moment - it requires information relating to the other tickets**

Check the GET_SCREENS PV:
```caget -S -t %MYPVPREFIX%CS:BLOCKSERVER:GET_SCREENS | uzhex```

For my system the PV value doesn't change - I don't know if this is my machine. If yours also doesn't update then it must be a bug. In that case, please create a ticket along the following lines:

```
Blockserver: GET_SCREENS PV does not update when setting screens
```

### Acceptance criteria

Can create and modify the devices screens via the GUI.

 **Before merging we need to disable the menu option as we don't want the users playing with it until the other hackathon tickets are ready.** 

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [ ] Are there [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has developer documentation been updated if required?

### Final Steps
- [x] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

John-Holt-Tessella commented:
1. The device screen name is not validated. What are the possible values. If I add the name `%^&<£` the block server crashes. But you can not tell it has crashed. The `waiting for the instrument to update` dialogue is shown and never goes. Not sure if the validation should be in here or in the block server or both. Either way crash of the block server should be fixed; it should die properly and get restarted.
1. Get as an error on save `INFO: The configuration, Eurotherm, has been modified in the filesystem, ensure it is added to version control`

mattclarke commented:
Added basic validation for names (1)
Renamed control (3)
Removed buttons which John and I think are redundant (4 and 5)
Table resizes correctly (6)
Dialog now expands correctly on resize drag (7)
Controls are disabled when nothing selected (8)

mattclarke commented:
2, 9 require tickets creating (I'll make these in the morning)
10, 11 are from a different dialog?
12 I agree, ticket?
13 as the dialog will be hidden until the whole "device screens" thing is finished the system tests probably should wait until then
